testing/e2e_cas_setup.sh

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env bash
+set -e
+set -o pipefail
+
+# $1 = expression
+# $2 = sleep period
+# $3 = iterations
+
+function wait_until() {
+  for _ in $(seq 1 "$3"); do
+    eval "$1" && return 0
+    echo "Waiting for \"$1\" to evaluate to true ..."
+    sleep "$2"
+  done
+  echo Timed out waiting for \""$1"\"
+  return 1
+}
+
+if [ -z "${GIT_USER}" ]; then
+  echo "Set the GIT_USER env variable"
+  exit
+fi
+
+if [ -z "${GIT_PASSWORD}" ]; then
+  echo "Set the GIT_PASSWORD env variable"
+  exit
+fi
+
+if [ -z "${GIT_EMAIL}" ]; then
+  echo "Set the GIT_EMAIL env variable"
+  exit
+fi
+
+# Check for variables required for creating CAPA clusters
+if [ -z "${AWS_REGION}" ]; then
+  echo "Set the AWS_REGION env variable"
+  exit
+fi
+
+if [ -z "${AWS_ACCESS_KEY_ID}" ]; then
+  echo "Set the AWS_ACCESS_KEY_ID env variable"
+  exit
+fi
+
+if [ -z "${AWS_SECRET_ACCESS_KEY}" ]; then
+  echo "Set the AWS_SECRET_ACCESS_KEY env variable"
+  exit
+fi
+
+if [ -z "${AWS_CONTROL_PLANE_MACHINE_TYPE}" ]; then
+  echo "Set the AWS_CONTROL_PLANE_MACHINE_TYPE env variable"
+  exit
+fi
+
+if [ -z "${AWS_NODE_MACHINE_TYPE}" ]; then
+  echo "Set the AWS_NODE_MACHINE_TYPE env variable"
+  exit
+fi
+
+if [ -z "${AWS_SSH_KEY_NAME}" ]; then
+  echo "Set the AWS_SSH_KEY_NAME env variable"
+  exit
+fi
+
+# Setting os for arlon binary download
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+  os="linux"
+#  arlon_os="Linux"
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+  os="darwin"
+#  arlon_os="Darwin"
+fi
+
+cpu=$(uname -m)
+if [[ "$cpu" == "x86_64" ]]; then
+  arlon_arch="x86_64"
+  arch="amd64"
+elif [[ "$cpu" == "arm64"* ]]; then
+  #  arlon_arch="arm64"
+  arch="arm64"
+fi
+if [ ! -d "$HOME/.local/bin" ]; then
+  mkdir -p "$HOME/.local/bin"
+fi
```

Comment:
```suggestion
```

Might not be needed

Reply:
We still require this to download kubectl in the setup script